wazuhapp/Docker/build.sh

```diff
@@ -2,83 +2,119 @@
 
 set -ex
 
+# Script parameters
 wazuh_branch=$1
 checksum=$2
 app_revision=$3
 
+# Paths
 kibana_dir="/tmp/source"
 source_dir="${kibana_dir}/plugins/wazuh"
 build_dir="${source_dir}/build"
 destination_dir="/wazuh_app"
 checksum_dir="/var/local/checksum"
-github_raw_url="https://raw.githubusercontent.com/wazuh/wazuh-kibana-app"
-package_json_url="${github_raw_url}/${wazuh_branch}/package.json"
 
-export package_json="${source_dir}/package.json"
+# Repositories URLs
+wazuh_app_clone_repo_url="https://github.com/wazuh/wazuh-kibana-app.git"
+wazuh_app_raw_repo_url="https://raw.githubusercontent.com/wazuh/wazuh-kibana-app"
+kibana_app_repo_url="https://github.com/elastic/kibana.git"
+kibana_app_raw_repo_url="https://raw.githubusercontent.com/elastic/kibana"
+wazuh_app_package_json_url="${wazuh_app_raw_repo_url}/${wazuh_branch}/package.json"
 
+# Script vars
 wazuh_version=""
 kibana_version=""
-node_version=""
+wazuh_app_node_version=""
+kibana_node_version=""
+kibana_yarn_version=""
+
+
+change_node_version () {
+    installed_node_version="$(node -v)"
+    node_version=$1
+
+    n ${node_version}
+
+    if [[ "${installed_node_version}" != "v${node_version}" ]]; then
+        mv /usr/local/bin/node /usr/bin
+        mv /usr/local/bin/npm /usr/bin
+        mv /usr/local/bin/npx /usr/bin
+    fi
+
+    echo "Using $(node -v) node version"
+}
+
 
 prepare_env() {
-    echo "Download package.json"
+    echo "Downloading package.json from wazuh-kibana app repository"
+    if ! curl $wazuh_app_package_json_url -o "/tmp/package.json" ; then
+        echo "Error downloading package.json from GitHub."
+        exit 1
+    fi
+
+    wazuh_version=$(python -c 'import json, os; f=open("/tmp/package.json"); pkg=json.load(f); f.close();\
+                    print(pkg["version"])')
+    kibana_version=$(python -c 'import json, os; f=open("/tmp/package.json"); pkg=json.load(f); f.close();\
+                     print(pkg["kibana"]["version"])')
+
+    kibana_package_json_url="${kibana_app_raw_repo_url}/v${kibana_version}/package.json"
 
-    if ! curl -O ${package_json_url} ; then
+    echo "Downloading package.json from elastic/kibana repository"
+    if ! curl $kibana_package_json_url -o "/tmp/package.json" ; then
         echo "Error downloading package.json from GitHub."
         exit 1
     fi
 
-    wazuh_version=$(python -c 'import json, os; f=open("/package.json"); pkg=json.load(f); f.close(); print(pkg["version"])')
-    kibana_version=$(python -c 'import json, os; f=open("/package.json"); pkg=json.load(f); f.close(); print(pkg["kibana"]["version"])')
+    kibana_node_version=$(python -c 'import json, os; f=open("/tmp/package.json"); pkg=json.load(f); f.close();\
```

Comment:
This variable is never used

Reply:
Indeed. This variable was used in the previous version where the Kibana repository dependencies were installed.

Reply:
:ok_hand: